oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -561,6 +561,12 @@ public static ServiceAccountCredentials fromStream(
             fileType, SERVICE_ACCOUNT_FILE_TYPE));
   }
 
+  /** Returns whether the scopes are empty, meaning createScoped must be called before use. */
+  @Override
+  public boolean createScopedRequired() {
+    return scopes.isEmpty();
```

Comment:
I think without having `defaultScopes.isEmpty()` here as well, the [Client Library for Storage](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/storage/json-api/src/main/java/StorageFactory.java#L50) will never use self-signed JWTs. And there may be more APIs like this.

This can be fixed in a future PR, however.